src/main/java/bio/terra/workspace/service/controlledresource/ControlledGoogleResourceService.java

```diff
@@ -0,0 +1,55 @@
+package bio.terra.workspace.service.controlledresource;
+
+import bio.terra.workspace.generated.model.CreateControlledGoogleBucketRequestBody;
+import bio.terra.workspace.generated.model.GoogleBucketCreationParameters;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobBuilder;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.workspace.flight.CreateControlledGoogleBucketFlight;
+import bio.terra.workspace.service.workspace.flight.GoogleBucketFlightMapKeys;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ControlledGoogleResourceService {
+
+  private final JobService jobService;
+
+  @Autowired
+  public ControlledGoogleResourceService(JobService jobService) {
+    this.jobService = jobService;
+  }
+
+  public String createBucket(
+      UUID workspaceId,
+      CreateControlledGoogleBucketRequestBody requestBody,
+      AuthenticatedUserRequest userRequest) {
+    // create a job
+    final String description =
+        "Create controlled Google bucket named " + requestBody.getGoogleBucket().getName();
+    final JobBuilder jobBuilder =
+        jobService.newJob(
+            description,
+            requestBody.getCommon().getJobControl().getId(),
+            CreateControlledGoogleBucketFlight.class,
+            requestBody,
+            userRequest);
+    final GoogleBucketCreationParameters params = requestBody.getGoogleBucket();
+    jobBuilder.addParameter(WorkspaceFlightMapKeys.WORKSPACE_ID, workspaceId);
+    jobBuilder.addParameter(
+        WorkspaceFlightMapKeys.IAM_OWNER_GROUP_EMAIL, userRequest.getEmail()); // IS THIS RIGHT?
```

Comment:
I think we want a new user email flight key

Reply:
You are passing the request body right into the flight, so can't you pull out everything you need from that? I'm confused.

Reply:
I had it the other way first. Is it OK to just give it the whole thing, or is that too much coupling?